src/variable.cpp

```diff
@@ -4374,6 +4375,43 @@ int Variable::special_function(char *word, char *contents, Tree **tree, Tree **t
       newtree->value = value;
       treestack[ntreestack++] = newtree;
     } else argstack[nargstack++] = value;
+
+  } else if (strcmp(word,"label2type") == 0) {
+    if (narg != 2) print_var_error(FLERR,"Invalid label2type() function syntax in variable formula",ivar);
+
+    if (!atom->labelmapflag)
+      print_var_error(FLERR,"Cannot use label2type() function without a labelmap",ivar);
+
+    int value;
+    std::string kind = args[0];
+    std::string typestr = utils::trim(utils::utf8_subst(args[1]));
```

Comment:
one last comment/clarification. the refactored label2type() variable function technically changes what a valid type label is. previously, type labels could contain commas, but this is no longer supported

Reply:
Technically, the limitation only applies to the variable function. There are other limitations there, too, when using immediate variable evaluation with `$(label2type(kind,label))`. The case of '.' can be addressed, but there are also the characters '(' and ')' that cannot be used with the variable function. Well, you *can* use something like '(A)' or '(A-B)' as label with the label2type() function since that has a pair of parenthesis.